src/AutoMapper/MemberMap.cs

```diff
@@ -39,7 +39,20 @@ public class MemberMap
         public virtual ValueResolverConfiguration ValueResolverConfig { get => default; set { } }
         public virtual ValueResolverConfiguration ValueConverterConfig { get => default; set { } }
         public virtual IReadOnlyCollection<ValueTransformerConfiguration> ValueTransformers => Array.Empty<ValueTransformerConfiguration>();
-        public MemberInfo SourceMember => CustomMapExpression.GetMember() ?? SourceMembers.LastOrDefault();
+        public MemberInfo SourceMember => this switch
+        {
```

Comment:
I think it would make a lot of sense to have only one resolver object per member map (we have 5 kinds of resolvers). It would break cases when people use a different configuration for `Map` and `ProjectTo`, but I guess that's a small price to pay :)